src/screens/communities/container/communitiesContainer.js

```diff
@@ -26,88 +30,90 @@ const CommunitiesContainer = ({ children, navigation }) => {
-            });
-          }
-        }
-      }
-    });
-
-    setSubscriptions(subscribedsData);
-  }, [subscribingCommunitiesInJoinedTab]);
+  }, [subscribedCommunities]);
+
+  //  Commented this code as it not serving any purpose now because we are handling data in redux
+  // useEffect(() => {
+  //   const discoversData = [...discovers];
+
+  //   Object.keys(subscribingCommunitiesInDiscoverTab).map((communityId) => {
+  //     if (!subscribingCommunitiesInDiscoverTab[communityId].loading) {
+  //       if (!subscribingCommunitiesInDiscoverTab[communityId].error) {
+  //         if (subscribingCommunitiesInDiscoverTab[communityId].isSubscribed) {
+  //           discoversData.forEach((item) => {
+  //             if (item.name === communityId) {
+  //               item.isSubscribed = true;
+  //             }
+  //           });
+  //         } else {
+  //           discoversData.forEach((item) => {
+  //             if (item.name === communityId) {
+  //               item.isSubscribed = false;
+  //             }
+  //           });
+  //         }
+  //       }
+  //     }
+  //   });
+
+  //   setDiscovers(discoversData);
+  // }, [subscribingCommunitiesInDiscoverTab]);
+
+  // useEffect(() => {
+  //   const subscribedsData = [...subscriptions];
+
+  //   Object.keys(subscribingCommunitiesInJoinedTab).map((communityId) => {
+  //     if (!subscribingCommunitiesInJoinedTab[communityId].loading) {
+  //       if (!subscribingCommunitiesInJoinedTab[communityId].error) {
+  //         if (subscribingCommunitiesInJoinedTab[communityId].isSubscribed) {
+  //           subscribedsData.forEach((item) => {
+  //             if (item[0] === communityId) {
+  //               item[4] = true;
+  //             }
+  //           });
+  //         } else {
+  //           subscribedsData.forEach((item) => {
+  //             if (item[0] === communityId) {
+  //               item[4] = false;
+  //             }
+  //           });
+  //         }
+  //       }
+  //     }
+  //   });
+
+  //   setSubscriptions(subscribedsData);
+  // }, [subscribingCommunitiesInJoinedTab]);
 
   const _getSubscriptions = () => {
-    setIsSubscriptionsLoading(true);
-    getSubscriptions(currentAccount.username)
-      .then((subs) => {
-        subs.forEach((item) => item.push(true));
-        getCommunities('', 50, null, 'rank').then((communities) => {
-          communities.forEach((community) =>
-            Object.assign(community, {
-              isSubscribed: subs.some(
-                (subscribedCommunity) => subscribedCommunity[0] === community.name,
-              ),
-            }),
-          );
-
-          setSubscriptions(subs);
-          setDiscovers(shuffle(communities));
-          setIsSubscriptionsLoading(false);
+    if (subscribedCommunities && subscribedCommunities.length > 0) {
+      setIsSubscriptionsLoading(true);
+      let subs = subscribedCommunities.slice();
+      getCommunities('', 50, null, 'rank').then((communities) => {
```

Comment:
This is probably causing a loading glitch when first entering communities screen, perhaps we should keep the two tabs data separate... show subscribed communities from redux cache separate, and fetch + post process ranked communities separate